core/core-api/src/main/java/com/hbc/pms/core/api/config/ModelMapperConfig.java

```diff
@@ -1,20 +1,94 @@
 package com.hbc.pms.core.api.config;
 
+import com.hbc.pms.core.api.controller.v1.request.UpdateSensorConfigurationRequest;
+import com.hbc.pms.core.api.controller.v1.response.BlueprintResponse;
+import com.hbc.pms.core.api.utils.Utils;
+import com.hbc.pms.core.model.Blueprint;
+import com.hbc.pms.core.model.SensorConfiguration;
+import org.modelmapper.Converter;
 import org.modelmapper.ModelMapper;
+import org.modelmapper.PropertyMap;
+import org.modelmapper.ValidationException;
 import org.modelmapper.convention.MatchingStrategies;
+import org.modelmapper.spi.ErrorMessage;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 @Configuration
 public class ModelMapperConfig {
-  @Bean
-  public ModelMapper modelMapper() {
-    ModelMapper modelMapper = new ModelMapper();
-    modelMapper
-        .getConfiguration()
-        .setMatchingStrategy(MatchingStrategies.STANDARD)
-        .setSkipNullEnabled(true)
-        .setDeepCopyEnabled(true);
-    return modelMapper;
-  }
+    @Bean
+    public ModelMapper modelMapper() throws ValidationException {

```

Comment:
looks really nice

Reply:
but can we do this another way to make this easier to read

Reply:
Please check my modifications. If there are still ways to optimize it, I would be grateful if you could briefly show me how.

Reply:
That's all I can do; the rest is due to the syntax of the ModelMapper, so I still can't figure out how to do it better.
If there are still ways to make this better, I would be grateful if you just briefly show me how